Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/EventAspectAdvice.java

```diff
@@ -47,123 +48,124 @@
-        adapterDelegationAdviceDelegate.end(joinPoint.getArgs(), annotation.serviceType(), annotation.version(),
-                annotation.afterReturningBuilder(), returnValue);
-    }
-
-    @Before("@annotation(annotation)")
-    public void beginOutboundMessageEvent(JoinPoint joinPoint, OutboundMessageEvent annotation) {
-        outboundMessageAdviceDelegate.begin(joinPoint.getArgs(), annotation.serviceType(), annotation.version(),
-                annotation.beforeBuilder());
-    }
-
-    @AfterReturning(pointcut = "@annotation(annotation)", returning = "returnValue")
-    public void endOutboundMessageEvent(JoinPoint joinPoint, OutboundMessageEvent annotation, Object returnValue) {
-        outboundMessageAdviceDelegate.end(joinPoint.getArgs(), annotation.serviceType(), annotation.version(),
-                annotation.afterReturningBuilder(), returnValue);
-    }
-
-    @Before("@annotation(annotation)")
-    public void beginOutboundProcessingEvent(JoinPoint joinPoint, OutboundProcessingEvent annotation) {
-        outboundProcessingAdviceDelegate.begin(joinPoint.getArgs(), annotation.serviceType(), annotation.version(),
-                annotation.beforeBuilder());
-    }
-
-    @AfterReturning(pointcut = "@annotation(annotation)", returning = "returnValue")
-    public void endOutboundProcessingEvent(JoinPoint joinPoint, OutboundProcessingEvent annotation,
-            Object returnValue) {
-        outboundProcessingAdviceDelegate.end(joinPoint.getArgs(), annotation.serviceType(), annotation.version(),
-                annotation.afterReturningBuilder(), returnValue);
-    }
-
-    @Before("@annotation(annotation)")
-    public void beginNwhinInvocationEvent(JoinPoint joinPoint, NwhinInvocationEvent annotation) {
-        nwhinInvocationAdviceDelegate.begin(joinPoint.getArgs(), annotation.serviceType(), annotation.version(),
-                annotation.beforeBuilder());
-    }
-
-    @AfterReturning(pointcut = "@annotation(annotation)", returning = "returnValue")
-    public void endNwhinInvocationEvent(JoinPoint joinPoint, NwhinInvocationEvent annotation, Object returnValue) {
-        nwhinInvocationAdviceDelegate.end(joinPoint.getArgs(), annotation.serviceType(), annotation.version(),
-                annotation.afterReturningBuilder(), returnValue);
-    }
-
-    /*
-     * Passthrough modes do not have any Event annotations on them. We want to support the ability to log exceptions
-     * for both Standard and Passthrough. If the method is annotated with both for some reason, we only process the
-     * LogFailures annotation to prevent recording duplicate entries for the failure.
+    @AfterThrowing(pointcut = "@annotation(gov.hhs.fha.nhinc.aspect.LogFailures)", throwing = "e")
+    public void logFailureAnnotation(JoinPoint joinPoint, Throwable e) {
+        logFailure(joinPoint, e, "Event Service", "0.0");
```

Comment:
Why there are default values used for calling logFailure? Is there a plan to replace them in next PR?

Reply:
This annotation is meant to be a generic failure that can be added onto any Spring bean to log a failure. We can talk about it a bit more at 3 PM today.